### ISSUE TITLE
Test our SVG implementation with resvg-test-suite

### ISSUE THREAD

**Issue by shlyakpavel** [LadybirdBrowser/ladybird#2306]
This looks like a great opportunity to benchmark against https://github.com/linebender/resvg-test-suite
It looks like that set of test has no runner or something, so implementing a script that runs our SVG renderer and compares it against reference pngs is up to us

**Comment by shlyakpavel** [comment]
1680 tests done
1600 if them are not UB
Final success rate for ladybird with all the tests: 27.5%
This is not bad at all. We also have extraordinary low crash rate. Feel free to take data on github and analyze it yourself https://github.com/linebender/resvg-test-suite/pull/55
![output-4](https://github.com/user-attachments/assets/0e6476f6-d0dc-4660-aaf8-e3dcdf46b00a)

**Comment by shlyakpavel** [comment]
In my opinion, top priority for fixing thesr tests are:
1) Gradients (not positioned correctly)
2) Font rendering
3) Asian characters (Japanese and Arabic)
4) Blur (I guess we don't support it at all)
5) Nested images (inline pngs jpegs etc)
6) Markers (I think we don't support these at all)
Other issues were very rare to occur

**Comment by gmta** [comment]
@shlyakpavel Do you have a simple reproduction case for one of the gradient mispositioning issues?

**Comment by shlyakpavel** [comment]
@gmta MDN has https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Gradients
 Safari:
 
<img width="864" alt="image" src="https://github.com/user-attachments/assets/48d6569e-e59b-4e57-ba05-e8e7685ef81f">

Ladybird:

<img width="701" alt="image" src="https://github.com/user-attachments/assets/66186037-8091-4af5-9fbb-b1cd8fb00343">

**Comment by gmta** [comment]
This is how that page looks on my machine:
![image](https://github.com/user-attachments/assets/b5a27c0d-61d3-485c-806c-ae1687c81207)

Can you still reproduce it with the latest `master`? What platform are you running on?

**Comment by gmta** [comment]
Ah, it does get mangled when you start to zoom in. I'm guessing this might be a difference in DPI

**Comment by shlyakpavel** [comment]
@gmta I have a 13 inch MacBook Air (m1) running macOS 15.1

**Comment by nico** [comment]
(Possibly a regression from moving to Skia; looks fine in Serenity's browser's mac port on retina.)

<img width="733" alt="image" src="https://github.com/user-attachments/assets/4891985c-60fc-4431-b662-06234b943656">

(This is the example in a standalone file; the full thing doesn't load in Serenity's browser apparently)

**Comment by gmta** [comment]
@nico No, you probably didn't cherry pick everything yet :-) this is the result of one of my changes in the way we deal with gradient coordinate spaces.
